src/components/cc-article-card/cc-article-card.js

```diff
@@ -21,59 +30,39 @@ export class CcArticleCard extends LitElement {
 
   static get properties () {
     return {
-      banner: { type: String },
-      date: { type: String },
-      description: { type: String },
-      link: { type: String },
-      title: { type: String },
+      state: { type: Object },
     };
   }
 
   constructor () {
     super();
 
-    /** @type {string|null} Sets the article banner image. */
-    this.banner = null;
-
-    /** @type {string|null} Sets the article date. */
-    this.date = null;
-
-    /** @type {string|null} Sets a short description of the article. */
-    this.description = null;
-
-    /** @type {string|null} Sets the link to the article. */
-    this.link = null;
-
-    /** @type {string|null} Sets the title of the article. */
-    this.title = null;
+    /** @type {ArticleCardState} Sets the state of the component */
+    this.state = { type: 'loading' };
   }
 
   render () {
 
-    const skeleton = (this.title == null || this.banner == null || this.link == null || this.date == null);
-
-    const banner = skeleton ? null : this.banner;
-    const title = this.title ?? SKELETON_INFO.title;
-    const description = this.description ?? SKELETON_INFO.description;
-    const date = this.date ?? SKELETON_INFO.date;
+    const skeleton = this.state.type === 'loading';
+    const data = this.state.type === 'loaded' ? this.state : SKELETON_INFO;
 
     return html`
-      <cc-img class="image" src=${banner}></cc-img>
+      <cc-img class="image" src=${ifDefined(data.banner)}></cc-img>
       ${skeleton ? html`
         <div class="title">
-          <span class="skeleton">${title}</span>
+          <span class="skeleton">${data.title}</span>
         </div>
       ` : ''}
-      ${!skeleton ? html`
+      ${this.state.type === 'loaded' ? html`
         <div class="title">
-          ${ccLink(this.link, title, skeleton)}
+          ${ccLink(this.state.link, data.title)}
```

Comment:
suggestion: while `this.state.link` is valid, you could use `data.link` to make it less confusing